Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -913,3 +916,407 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
+                        lua_pushboolean(luaVM, true);
+                        return 1;
+                    }
+                    break;
+                case SURFACE_PROPERTY_STAIRS:
+                    argStream.ReadBool(bEnabled);
+                    if (!argStream.HasErrors())
+                    {
+                        pSurface->setFlagEnabled(0, 20, bEnabled);
+                        lua_pushboolean(luaVM, true);
+                        return 1;
+                    }
+                    break;
+                case SURFACE_PROPERTY_ROUGHNESS:
+                    char cRoughness;
+                    argStream.ReadNumber(cRoughness);
+                    if (!argStream.HasErrors() && cRoughness >= 0 && cRoughness <= 3)
+                    {
+                        pSurface->m_roughness = cRoughness;
+                        lua_pushboolean(luaVM, true);
+                        return 1;
+                    }
+                    break;
+            }
+        }
+        else
+            argStream.SetCustomError("Expected valid surface ID ( 0 - 179 ) at argument 1");
+    }
+    if (argStream.HasErrors())
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    lua_pushboolean(luaVM, false);
+    return 1;
+}
+
+const char* cSurfaceAudio[8] = {"concrete", "grass", "sand", "gravel", "wood", "water", "metal", "longGrass"};
+const char* cSurfaceStepEffect[2] = {"sand", "water"};
+const char* cSurfaceBulletEffect[4] = {"metal", "sand", "wood", "concrete"};
+const char* cSurfaceWheelEffect[5] = {"grass", "gravel", "mud", "sand", "dust"};
+const char* cSurfaceSkidMark[3] = {"asphalt", "dirt", "dust"};
+const char* cSurfaceAdhesionGroup[6] = {"rubber", "hard", "road", "loose", "sand", "wet"};
+
+int CLuaEngineDefs::EngineGetSurfaceProperties(lua_State* luaVM)
+{
+    int                iSurfaceID;
+    eSurfaceProperties eType;
+    CScriptArgReader   argStream(luaVM);
+    argStream.ReadNumber(iSurfaceID);
+    argStream.ReadEnumString(eType);
+    if (!argStream.HasErrors())
+    {
+        if (iSurfaceID >= EColSurfaceValue::DEFAULT && iSurfaceID <= EColSurfaceValue::SIZE)
+        {
+            CSurfaceType*  pSurfaceInfo = g_pGame->GetWorld()->GetSurfaceInfo();
+            SurfaceInfo_c* pSurface = &pSurfaceInfo->surfType[iSurfaceID];
+            switch (eType)
+            {
+                case SURFACE_PROPERTY_AUDIO:
+                    for (char cFlag = SURFACE_AUDIO_CONCRETE; cFlag <= SURFACE_AUDIO_METAL; cFlag++)
+                    {
+                        if (pSurface->getFlagEnabled(1, cFlag))
+                        {
+                            lua_pushstring(luaVM, cSurfaceAudio[cFlag - SURFACE_AUDIO_CONCRETE]);
+                            return 1;
+                        }
+                    }
+                    lua_pushstring(luaVM, "disabled");
+                    return 1;
+                    break;
```

Comment:
`break;` not required after `return 1;`. **Remove it for the other `case X:`s too!**